src/Symfony/Component/Form/Extension/Core/DataTransformer/MoneyToLocalizedStringTransformer.php

```diff
@@ -69,8 +69,7 @@ public function reverseTransform(mixed $value): int|float|null
         $value = parent::reverseTransform($value);
         if (null !== $value) {
             $value = (string) ($value * $this->divisor);
-
-            if ('float' === $this->input) {
+            if (\in_array($this->input, ['float', 'string'], true)) {
```

Comment:
This looks wrong to me. If the initial input is a string, the model data after the form has been submitted has to be a string as well.